python/amici/parameter_mapping.py

```diff
@@ -0,0 +1,368 @@
+        in the parameter mapping. If False, they
+        are assumed to be in linear scale.
+    :param parameter_mapping:
+        Parameter mapping for current condition.
+    :param amici_model:
+        AMICI model
+    """
+    map_sim_var = parameter_mapping.map_sim_var
+    scale_map_sim_var = parameter_mapping.scale_map_sim_var
+    map_preeq_fix = parameter_mapping.map_preeq_fix
+    scale_map_preeq_fix = parameter_mapping.scale_map_preeq_fix
+    map_sim_fix = parameter_mapping.map_sim_fix
+    scale_map_sim_fix = parameter_mapping.scale_map_sim_fix
+
+    # Parameter mapping may contain parameter_ids as values, these *must*
+    # be replaced
+
+    def _get_par(model_par, value):
+        """Replace parameter IDs in mapping dicts by values from
+        problem_parameters where necessary"""
+        if isinstance(value, str):
+            # estimated parameter
+            # (condition table overrides must have been handled already,
+            # e.g. by the PEtab parameter mapping)
+            return problem_parameters[value]
+        if model_par in problem_parameters:
+            # user-provided
+            return problem_parameters[model_par]
+        # constant value
+        return value
+
+    map_preeq_fix = {key: _get_par(key, val)
+                     for key, val in map_preeq_fix.items()}
+    map_sim_fix = {key: _get_par(key, val)
+                   for key, val in map_sim_fix.items()}
+    map_sim_var = {key: _get_par(key, val)
+                   for key, val in map_sim_var.items()}
+
+    # If necessary, (un)scale parameters
+    if scaled_parameters:
+        unscale_parameters_dict(map_preeq_fix, scale_map_preeq_fix)
+        unscale_parameters_dict(map_sim_fix, scale_map_sim_fix)
+    if not scaled_parameters:
+        # We scale all parameters to the scale they are estimated on, and pass
+        # that information to amici via edata.{parameters,pscale}.
+        # The scaling is necessary to obtain correct derivatives.
+        scale_parameters_dict(map_sim_var, scale_map_sim_var)
+        # We can skip preequilibration parameters, because they are identical
+        # with simulation parameters, and only the latter are used from here
+        # on.
+
+    ##########################################################################
+    # variable parameters and parameter scale
+
+    # parameter list from mapping dict
+    parameters = [map_sim_var[par_id]
+                  for par_id in amici_model.getParameterIds()]
+
+    # scales list from mapping dict
+    scales = [petab_to_amici_scale(scale_map_sim_var[par_id])
+              for par_id in amici_model.getParameterIds()]
+
+    if parameters:
+        edata.parameters = parameters
+
+    if scales:
+        edata.pscale = amici.parameterScalingFromIntVector(scales)
+
+    ##########################################################################
+    # fixed parameters preequilibration
+    if map_preeq_fix:
```

Comment:
```suggestion
    if len(map_preeq_fix):
```
`if {}:` will always pass

same below

Reply:
``if {}`` is actually False, same for empty lists ...

Reply:
no `len` is preferred. 

pylint: 
>
len-as-condition (C1801):
--
  | Do not use `len(SEQUENCE)` without comparison to determine if a sequence is empty Used when Pylint detects that len(sequence) is being used without explicit comparison inside a condition to determine if a sequence is empty. Instead of coercing the length to a boolean, either rely on the fact that empty sequences are false or compare the length against a scalar.

Reply:
hmm that pep actually states we should either do `if len(map_preeq_fix) > 0:` or `if map_preeq_fix:`. Wasn't aware that empty sequences are false, so fine with me.